flink-rpc/flink-rpc-core/src/main/java/org/apache/flink/runtime/rpc/MainThreadValidatorUtil.java

```diff
@@ -45,11 +45,13 @@ public MainThreadValidatorUtil(RpcEndpoint endpoint) {
     public void enterMainThread() {
         assert (endpoint.currentMainThread.compareAndSet(null, Thread.currentThread()))
                 : "The RpcEndpoint has concurrent access from " + endpoint.currentMainThread.get();
+        endpoint.beforeInvocation();
```

Comment:
We don't need both this and the main thread executor. All the main thread executor does is throw things into the rpc actor.

I also don't like this being coupled into the main thread validation logic; and would rather see this being handled in the PekkoRpcActor or encoded into runnables within the PekkoInvocationHandler.

Reply:
When I remove it, I get log lines like this without a job ID:
```
[                                ] ExecutionGraph [flink-pekko.actor.default-dispatcher-6] INFO  Source: Custom Source -> Timestamps/Watermarks -> transform-1-forward -> Sink: Unnamed (1/4) (2dae9a492649d761d992295551f74d3c_cbc357ccb763df2852fee8c4fc7d55f2_0_0) switched from SCHEDULED to DEPLOYING.
```

Reply:
Yes, because the main thread executable covers less; it doesnt capture anything sent directly against the rpc service's executor.